bounded-core/src/main/scala/io/cafienne/bounded/aggregate/AggregateRootActor.scala

```diff
@@ -35,10 +35,17 @@ trait AggregateRootActor extends PersistentActor with AggregateStateCreator with
     * When extending the AggregateRootActor you MUST implement the handleCommand method.
     *
     * @param command one of the commands your aggregate root expects
-    * @param state of the Aggregate Root at the moment of handling command
+    * @param state of the Aggregate Root at the moment of handling command, optional: state could not be set yet
     * @return a sequence of events when everything is Right, or an Failure (Left)
     */
-  def handleCommand(command: DomainCommand, state: AggregateState): Reply
+  def handleCommand(command: DomainCommand, state: Option[A]): Reply
+
+  private def handleCommand(command: DomainCommand, state: AggregateState): Reply = {
```

Comment:
this private method should not be necessary ? it is called by receiveCommand and that can call it without transformation by passing the Option of the state

Reply:
the ```def state: Option[AggregateState] ``` van be typed to A in order to ensure that the state is strongly typed

Reply:
And the AggregateStateCreator should return the specific typed state.